### DISCUSSION TITLE
Is there a simple mechanism for initializing a pydantic model using it's field examples as fixture data?

### DISCUSSION THREAD

**Discussion by richard-thatch** [pydantic/pydantic#4317] [Question]
I'm trying to streamline some of the fixtures for pydantic models we use in our codebase.

Consider the following example code.

```python
from pydantic import BaseModel, Field, EmailStr


class UserCreate(BaseModel):
    given_name: str = Field(..., example="Jane")
    family_name: str = Field(..., example="Doe")
    email: EmailStr = Field(..., example="test@example.com")


user = UserCreate(given_name="Jane", family_name="Doe", email="test@example.com")
```

If we endeavor to make required fields have "good and valid" example data, Instead of having to create these models ad hoc like how we are doing so in this example, is there any mechanism to ask Pydantic to "give me an example model" automatically given that the required fields all have examples for the purpose of the schema?

I couldn't find any example of this, it might not exist. If it doesn't, should it? Something a la `user = UserCreate.example_factory()`

This is a very minor annoyance, it can be argued that the explicit use of the model is more clear from a code completeness perspective.

I apologize if this has been raised at all before. I did look before asking.

**Comment by PrettyWood** [comment]
Hi! Try
```py
@classmethod
def get_example(cls):
    return cls(**{field_name: field.field_info.extra['example'] for field_name, field in cls.__fields__.items()})
 ```
